packages/components/organisms/f-checkout/src/components/tests/Checkout.test.js

```diff
@@ -619,38 +754,210 @@ describe('Checkout', () => {
+                    i18n,
+                    localVue,
+                    propsData,
+                    mocks: {
+                        $v
+                    }
+                });
+
+                // Act
+                await wrapper.vm.onFormSubmit();
+
+                // Assert
+                expect(isFormValidSpy).toBeCalled();
+                expect(getFormValidationStateSpy).toBeCalledWith($v);
+                expect(wrapper.emitted(EventNames.CheckoutFailure).length).toBe(1);
+                expect(wrapper.emitted(EventNames.CheckoutFailure)[0][0]).toEqual(mockValidationState);
+            });
+
+            it('should try to call `submitCheckout` if form is Valid', async () => {
+                // Arrange
+                isFormValidSpy.mockReturnValue(true);
+
+                const wrapper = mount(VueCheckout, {
+                    store: createStore(),
+                    i18n,
+                    localVue,
+                    propsData,
+                    mocks: {
+                        $v
+                    }
+                });
+
+                // Act
+                await wrapper.vm.onFormSubmit();
+
+                // Assert
+                expect(submitCheckoutSpy).toHaveBeenCalled();
+            });
+
+            it('should call `handleErrorState` if `submitCheckout` returns an error', async () => {
+                // Arrange
+                const handleErrorStateSpy = jest.spyOn(VueCheckout.methods, 'handleErrorState');
+                const error = new Error('errorMessage');
+
+                submitCheckoutSpy.mockImplementation(async () => { throw error; });
+                isFormValidSpy.mockReturnValue(true);
+
+                const wrapper = mount(VueCheckout, {
+                    store: createStore(),
+                    i18n,
+                    localVue,
+                    propsData,
+                    mocks: {
+                        $v
+                    }
+                });
+
+                // Act
+                await wrapper.vm.onFormSubmit();
+
+                // Assert
+                expect(handleErrorStateSpy).toHaveBeenCalledWith(error);
+            });
+        });
+
+        describe('isValidPhoneNumber ::', () => {
+            afterEach(() => {
+                jest.clearAllMocks();
+            });
+
+            it('should call `isValidPhoneNumber` from `f-services`', () => {
+                // Arrange
+                const isValidPhoneNumberSpy = jest.spyOn(validations, 'isValidPhoneNumber');
+
+                const wrapper = mount(VueCheckout, {
                     store: createStore(),
                     i18n,
                     localVue,
                     propsData
                 });
 
+                // Act
+                wrapper.vm.isValidPhoneNumber();
+
                 // Assert
                 expect(isValidPhoneNumberSpy).toBeCalledWith(defaultState.customer.mobileNumber, i18n.locale);
```

Comment:
Not your change but Is this call intentional? It's the same as `toHaveBeenCalledWith`

Reply:
Changed it, thanks